DMCompiler/Program.cs

```diff
@@ -4,163 +4,183 @@
-                        break;
-                    case "pragma-config": {
-                            if(arg.Value is null || !HasValidDMExtension(arg.Value)) {
-                                if(skipBad) {
-                                    DMCompiler.ForcedWarning($"Compiler arg 'pragma-config' requires filename of valid DM file, skipping");
-                                    continue;
-                                }
-                                Console.WriteLine("Compiler arg 'pragma-config' requires filename of valid DM file");
-                                return false;
-                            }
-                            settings.PragmaFileOverride = arg.Value;
-                            break;
-                    }
-                    case "version": {
-                        if(arg.Value is null) {
-                            if(skipBad) {
-                                DMCompiler.ForcedWarning("Compiler arg 'version' requires a full BYOND build (e.g. --version=514.1584), skipping");
-                                continue;
-                            }
-                            Console.WriteLine("Compiler arg 'version' requires a full BYOND build (e.g. --version=514.1584)");
-                            return false;
-                        }
+    private static bool HasValidDMExtension(string filename) {
+        var extension = Path.GetExtension(filename);
+        return !string.IsNullOrEmpty(extension) && (extension == ".dme" || extension == ".dm");
+    }
 
-                        var split = arg.Value.Split('.', StringSplitOptions.RemoveEmptyEntries);
-                        if (split.Length != 2 || !int.TryParse(split[0], out _) || !int.TryParse(split[1], out _)) { // We want to make sure that they *are* ints but the preprocessor takes strings
-                            if(skipBad) {
-                                DMCompiler.ForcedWarning("Compiler arg 'version' requires a full BYOND build (e.g. --version=514.1584), skipping");
-                                continue;
-                            }
-                            Console.WriteLine("Compiler arg 'version' requires a full BYOND build (e.g. --version=514.1584)");
-                            return false;
-                        }
+    private static void PrintHelp() {
+        Console.WriteLine("DM Compiler for OpenDream");
+        Console.WriteLine("For more information please visit https://github.com/OpenDreamProject/OpenDream/wiki");
+        Console.WriteLine("Usage: ./DMCompiler [options] [file].dme\n");
+        Console.WriteLine("Options and arguments:");
+        Console.WriteLine("--help                    : Show this help");
+        Console.WriteLine("--version                 : Print the compiler version");
+        Console.WriteLine("--skip-bad-args           : ");
+        Console.WriteLine("--suppress-unimplemented  : Do not log about unimplemented functions in OpenDream");
```

Comment:
This includes unimplemented vars, and there's no need to specify it's in OpenDream.
```suggestion
        Console.WriteLine("--suppress-unimplemented  : Do not warn about unimplemented proc and var uses");
```